homeassistant/components/keba/__init__.py

```diff
@@ -0,0 +1,212 @@
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+_SERVICE_MAP = {
+    'request_data': 'request_data',
+    'set_energy': 'async_set_energy',
+    'set_current': 'async_set_current',
+    'authorize': 'async_start',
+    'deauthorize': 'async_stop',
+    'enable': 'async_enable_ev',
+    'disable': 'async_disable_ev',
+    'set_failsafe': 'async_set_failsafe'
+}
+
+
+async def async_setup(hass, config):
+    """Check connectivity and version of KEBA charging station."""
+    host = config[DOMAIN][CONF_HOST]
+    rfid = config[DOMAIN]['rfid']
+    refresh_interval = config[DOMAIN]['refresh_interval']
+    keba = KebaHandler(hass, host, rfid, refresh_interval)
+    hass.data[DOMAIN] = keba
+
+    # Wait for KebaHandler setup complete (initial values loaded)
+    await keba.setup()
+
+    # Set failsafe mode at start up of home assistant
+    failsafe = config[DOMAIN]['failsafe']
+    timeout = config[DOMAIN]['failsafe_timeout'] if failsafe else 0
+    fallback = config[DOMAIN]['failsafe_fallback'] if failsafe else 0
+    persist = config[DOMAIN]['failsafe_persist'] if failsafe else 0
+    try:
+        hass.loop.create_task(keba.set_failsafe(timeout, fallback, persist))
+    except ValueError as ex:
+        _LOGGER.warning("Could not set failsafe mode %s", ex)
+
+    # Register services to hass
+    async def async_execute_service(call):
+        """Execute a service to KEBA charging station.
+
+        This must be a member function as we need access to the hass
+        object here.
+        """
+        function_name = _SERVICE_MAP[call.service]
+        function_call = getattr(keba, function_name)
+        hass.async_create_task(function_call(call.data))
+
+    for service in _SERVICE_MAP:
+        hass.services.async_register(DOMAIN, service, async_execute_service)
+
+    # Load components
+    for domain in SUPPORTED_COMPONENTS:
+        hass.async_create_task(
+            discovery.async_load_platform(hass, domain, DOMAIN, {}, config))
+
+    # Start periodic polling of charging station data
+    keba.start_periodic_request()
+
+    return True
+
+
+class KebaHandler(KebaKeContact):
+    """Representation of a KEBA charging station connection."""
+
+    def __init__(self, hass, host, rfid, refresh_interval):
+        """Constructor."""
+        super().__init__(host, self.hass_callback)
+
+        self._update_listeners = []
+        self._hass = hass
+        self.rfid = rfid
+        self.device_name = "keba_wallbox_"
+
+        # Ensure at least 5 seconds delay
+        self._refresh_interval = max(5, refresh_interval)
+        self._fast_polling_count = 9
+        self._polling_task = None
+
+    def start_periodic_request(self):
+        """Start periodic data polling."""
+        self._polling_task = self._hass.loop.create_task(
+            self._periodic_request()
+        )
+
+    async def _periodic_request(self):
+        """Send update requests asyncio style."""
+        self._hass.async_create_task(self.request_data())
```

Comment:
Why not await this coroutine directly?

Reply:
you are right, does not make sense, fixed with await self.request_data()